src/Altinn.Platform/Altinn.Platform.Storage/Storage/Controllers/DataController.cs

```diff
@@ -74,56 +83,59 @@ public async Task<IActionResult> Delete(Guid instanceId, string formId, Guid dat
             }
 
             // check if instance id exist and user is allowed to change the instance data            
             Instance instance = await _instanceRepository.GetOneAsync(instanceId, instanceOwnerId);
             if (instance == null)
             {
-                return NotFound("Provided instanceId is unknown to platform storage service");
+                return NotFound("Provided instanceId and instanceOwnerId is unknown to platform storage service");
             }
 
-            string storageFileName = DataFileName(instance.ApplicationId, instanceId.ToString(), formId, dataId.ToString());
+            string storageFileName = DataFileName(instance.ApplicationId, instanceId.ToString(), dataId.ToString());
+            string dataIdString = dataId.ToString();
 
             // check if dataId exists in instance
-            if (instance.Data.ContainsKey(formId))
+            if (instance.Data.ContainsKey(dataIdString))
             {
-                Dictionary<string, Data> formData = instance.Data[formId];
-                if (formData.ContainsKey(dataId.ToString()))
+                Data data = instance.Data[dataIdString];
+
+                if (string.Equals(data.StorageUrl, storageFileName))
                 {
-                    Data data = formData[dataId.ToString()];
+                    Stream dataStream = await _dataRepository.GetDataInStorage(storageFileName);
 
-                    if (string.Equals(data.StorageUrl, storageFileName))
+                    if (dataStream == null)
                     {
-                        Stream dataStream = await _dataRepository.GetDataInStorage(storageFileName);
-
-                        if (dataStream == null)
-                        {
-                            return NotFound();
-                        }
-
-                        return File(dataStream, data.ContentType, data.FileName);
+                        return NotFound("Unable to read data storage for " + dataIdString);
                     }
+
+                    return File(dataStream, data.ContentType, data.FileName);
                 }
             }
 
             return NotFound("Unable to find requested data item");
         }
 
-        public static string DataFileName(string ApplicationId, string instanceId, string formId, string dataId)
+        /// <summary>
+        /// Formats a filename for blob storage.
+        /// </summary>
+        /// <param name="ApplicationId">the application id</param>
+        /// <param name="instanceId">the instance id</param>
+        /// <param name="dataId">the data id</param>
+        /// <returns></returns>
+        public static string DataFileName(string ApplicationId, string instanceId, string dataId)
         {
-            return ApplicationId + "/" + instanceId + "/data/" + formId + "/" + dataId;
+            return ApplicationId + "/" + instanceId + "/data/" + dataId;
```

Comment:
Consider using string interpolation

Reply:
Ok. Have changed to string interpolation in most code lines